### ISSUE TITLE
Expand indexing error causes incorrect results

### ISSUE THREAD

**Issue by rdspring1** [csarofeen/pytorch#1514]
### 🐛 Expand indexing error causes incorrect results

It seems we are indexing into the expanded tensor incorrectly. The repro works correctly when the expand is dropped and we treat the size-1 axis as a broadcast dimension.

```python
import torch
inps = [(torch.Size([2, 128]), torch.float32), (torch.Size([2, 1]), torch.float32), (torch.Size([2, 128]), torch.float32)]
inps = [torch.empty(())] + [torch.randn(shape, dtype=dtype, device='cuda') for (shape, dtype) in inps]

def forward(self, sub, div_1, mul_8):
    expand = torch.ops.aten.expand(div_1, [2, 128]);  div_1 = None
    mul_10 = torch.ops.aten.mul(expand, sub);  expand = sub = None
    add = torch.ops.aten.add(mul_10, mul_8);  mul_10 = mul_8 = None
    return (add,)

ref = forward(*inps)
f = torch.jit.script(forward)
with torch.jit.fuser("fuser2"):
  for _ in range(5):
    res = f(*inps)

for a, b in zip(ref, res):
  print((a - b).abs().max())
```

### Versions

```
PyTorch version: 1.11.0a0+git7b068db
Is debug build: False
CUDA used to build PyTorch: 11.1
ROCM used to build PyTorch: N/A

OS: Ubuntu 20.04.2 LTS (x86_64)
GCC version: (Ubuntu 9.3.0-17ubuntu1~20.04) 9.3.0
Clang version: Could not collect
CMake version: version 3.18.2
Libc version: glibc-2.31

Python version: 3.8.8 (default, Apr 13 2021, 19:58:26)  [GCC 7.3.0] (64-bit runtime)
Python platform: Linux-5.4.0-70-generic-x86_64-with-glibc2.10
Is CUDA available: True
CUDA runtime version: 11.1.105
GPU models and configuration:
GPU 0: A100-SXM4-40GB
GPU 1: A100-SXM4-40GB
GPU 2: A100-SXM4-40GB
GPU 3: A100-SXM4-40GB

Nvidia driver version: 460.32.03
cuDNN version: Could not collect
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

Versions of relevant libraries:
[pip3] mypy==0.812
[pip3] mypy-extensions==0.4.3
[pip3] numpy==1.22.2
[pip3] numpydoc==1.1.0
[pip3] nvidia-dlprof-pytorch-nvtx==1.1.0
[pip3] torch==1.12.0a0+git5f7ab32
[pip3] torchaudio==0.8.1
[pip3] torchvision==0.9.1
[conda] Could not collect
```

**Comment by naoyam** [comment]
At the time of lowering, here's what the fusion looks like:

```
Inputs:
  T0_g[ iS0{i1}, iS1{i2} ], float
  T1_g[ iS2{i3}, iS3{i4} ], float
  T2_g[ iS4{i5}, iS5{i6} ], float
Outputs:
  T4_g[ iS8{i3}, iS9{i4} ], float

%kernel_math {
T3_l[ iS6{i3}, iS7{i4} ]
   = T1_g[ iS2{i3}, iS3{i4} ]
   * T2_g[ iS4{i5}, iS5{i6} ];
T4_g[ iS8{i3}, iS9{i4} ]
   = T3_l[ iS6{i3}, iS7{i4} ]
   + T0_g[ iS0{i1}, iS1{i2} ];
}
```

So, the fusion just have three 2-D tensors and does only pointwise ops with them, and there's no broadcast.

At the launch time, the kernel arguments are

```
Arguments for kernel1:
Inputs:
Float [2, 128]
Float [2, 128]
Float [2, 128]
Outputs:
  Float [2, 128]
Reduction and semaphore buffers:
```

This seems to match with what the kernel expects, so nothing seems wrong as far as lowering is concerned.

Is the above result expected? I'm not sure what the `torch.ops.aten.expand` should be translated to, but isn't it supposed to have a broadcast somewhere?

**Comment by naoyam** [comment]
Here's the actual generated code:

```
__global__ void kernel1(Tensor<float, 2> T0, Tensor<float, 2> T1, Tensor<float, 2> T2, Tensor<float, 2> T4) {
  int64_t i83;
  i83 = (((nvfuser_index_t)blockIdx.x) * 128) + ((nvfuser_index_t)threadIdx.x);
  if ((i83 < (T0.size[0] * T0.size[1]))) {
    float T5[1];
    T5[0] = 0;
    T5[0]
       = T0[i83];
    float T7[1];
    T7[0] = 0;
    T7[0]
       = T2[i83];
    float T6[1];
    T6[0] = 0;
    T6[0]
       = T1[((((((nvfuser_index_t)blockIdx.x) * 128) + ((nvfuser_index_t)threadIdx.x)) / T0.size[1]) * (T1.stride[1] * T0.size[1])) + ((((((nvfuser_index_t)blockIdx.x) * 128) + ((nvfuser_index_t)threadIdx.x)) % T0.size[1]) * T1.stride[1])];
    float T8[1];
    float T3[1];
    T3[0]
      = T6[0]
      * T7[0];
    T8[0]
      = T3[0]
      + T5[0];
    T4[i83]
       = T8[0];
  }
}
```

Everything looks just fine. The only rather unusual expression is the stride of the first axis of `T1`: `(T1.stride[1] * T0.size[1])`. That's because the contiguity vector of `T1` is set as `[true, false]`. So, as far as I can see, nothing seems suspicious with lowering.

**Comment by jjsjann123** [comment]
> Everything looks just fine. The only rather unusual expression is the stride of the first axis of T1: (T1.stride[1] * T0.size[1]). That's because the contiguity vector of T1 is set as [true, false]. So, as far as I can see, nothing seems suspicious with lowering.

I think T1 is [2, 1] expanded to [2, 128]? In this case, second dimension should be marked as broadcasted.... Let me double check the parser.

**Comment by jjsjann123** [comment]
I think this is because of some changes I made to stride indices to match eager behavior, which unfortunately messes up order of broadcasting dimensions...
We just need to be a little bit cautious of marking contiguous dimensions in integration. Let me get a quick patch for this.
